ui/scripts/vpc.js

```diff
@@ -1333,35 +1321,65 @@
-                               cidrlist: '10.1.1.0/24',
-                               protocol: 'TCP',
-                               startport: 22, endport: 22,
-                               networkid: 0,
-                               traffictype: 'Egress'
-                               },
-                               {
-                               cidrlist: '10.2.1.0/24',
-                               protocol: 'UDP',
-                               startport: 56, endport: 72,
-                               networkid: 0,
-                               trafficType: 'Ingress'
-                               }
-                               ]*/
                                                     });
+                                                    if(jQuery('#details-tab-aclRules').siblings('div.toolbar').children('div.add').size() === 0){
+                                                        var $addAclRuleDivButton = jQuery('<div>').addClass('button add');
+                                                        var $spanAddAclRuleButtonMessage = jQuery('<span>').html(_l('label.add.ACL'));
+                                                        
+                                                        $addAclRuleDivButton.html($spanAddAclRuleButtonMessage);
+                                                        $addAclRuleDivButton.click(function(){
+                                                        	cloudStack.dialog.createForm({
+                                                        		 form: {
+                                                                     title: 'label.add.rule',
+                                                                     desc: 'Create a new ACL rule',
+                                                                     fields: aclRuleFields
+                                                                 },
+                                                                 after: function(argsLocal) {
+                                                                	 var data = argsLocal.data;
+                                                                	 data.aclid = argsLocal.context.aclLists[0].id;
+                                                                	 if(data.protocol != 'icmp'){
+                                                                		 data.icmpcode = undefined;
+                                                                		 data.icmptype  = undefined;
+                                                                	 }
+                                                                	 if(data.protocol != 'protocolnumber'){
+                                                                		 data.protocolnumber = undefined;
+                                                                	 }
+                                                                	 if(data.protocol === 'all'){
```

Comment:
Shouldn't be safer to perform these checks on the server side depending on the protocol parameter for createNetworkACL API method instead of undefining parameters to be sent?

Reply:
Yes, and they are in fact already performed there. The point is that if some of these combinations are sent to the server side, such as ICMP codes and TCP protocol, the server side will throw an exception. That is why I am cleaning up undesired information here (based on the protocol selected).

Reply:
Got it. Thanks for clarifying this